src/roman_datamodels/datamodels.py

```diff
@@ -6,23 +6,29 @@
 
 It is to be subclassed by the various types of data model variants for products
 """
+
+import contextlib
 import copy
 import datetime
+import logging
 import os
-import os.path
+import os.path as op
+import re
 import sys
 import warnings
 from collections import OrderedDict
 from collections.abc import Iterable
-from pathlib import PurePath
+from pathlib import Path, PurePath
 
 import asdf
 import numpy as np
 import packaging.version
 from astropy.time import Time
+from romancal.associations import AssociationNotValidError, load_asn
```

Comment:
We can't have circular references like this. 
Romancal depends on roman_datamodels, the opposite should not be true.